javascript-client/src/client.ts

```diff
@@ -140,15 +186,18 @@ export default class S3FileFieldClient {
   protected async completeUpload(
     multipartInfo: MultipartInfo, parts: UploadedPart[],
   ): Promise<void> {
-    const response = await this.api.post('upload-complete/', {
+    const response = await retry<AxiosResponse>(() => this.api.post('upload-complete/', {
```

Comment:
Since POST is not necessarily idempotent, do we know what happens if a client re-calls the endpoint mistakenly (perhaps due to the network dropping only the response)?

If this endpoint is actually idempotent, maybe we should change it to a PUT request.

Reply:
I think either the endpoint should be idempotent, or if not, it should return 400 if a duplicate happens, which would break us out of the loop.